### PR TITLE
GenericNodeTransformation - checking validators during steps evaluati…

### PR DESCRIPTION
**Pull request by arkadius**
…on instead of at the end - it was too late because implementation was invoked but it shouldn't be

### CODE REVIEW

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala [hunk 1]
```diff
@@ -56,14 +57,21 @@ object Validations {
...
-    val validationResults = paramWithValidatorList.map {
-      case (param, validator) => validator.isValid(param.name, param.expression.expression, None).toValidatedNel
-    }
+      paramDefinition <- definitionsMap.get(param.name)
+      paramValidationResult = validateParameterWithCustomValidators(paramDefinition, param)
+    } yield paramValidationResult
     validationResults.sequence.map(_ => Unit)
   }
+
+  def validateParameterWithCustomValidators[T >: PartSubGraphCompilationError <: ProcessCompilationError](paramDefinition: Parameter,
```
Comment (mproch) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala:69]: I wonder, could it be a method in Parameter? Or do we want to keep them anemic ;) 
Also, I think 'custom' is misleading here - they are in most cases rather standard?

File: utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/ConfluentSchemaRegistryClient.scala [hunk 2]
```diff
@@ -29,6 +32,35 @@ trait ConfluentSchemaRegistryClient extends SchemaRegistryClient with LazyLoggin
     }
 }
 
+class DefaultConfluentSchemaRegistryClient(override val client: CSchemaRegistryClient) extends ConfluentSchemaRegistryClient {
```
Comment (mproch) [inline utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/ConfluentSchemaRegistryClient.scala:35]: I don't think I fully understand why is this needed for this change?

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala [hunk 3]
```diff
@@ -21,13 +21,22 @@ object StandardParameterEnrichment {
...
+    val finalValidators = determineValidatorss(original, parameterConfig, parameterData, finalEditor)
     val isOptional = original.isOptional
     val finalDefaultValue = original.defaultValue.orElse(DefaultValueDeterminerChain.determineParameterDefaultValue(
       DefaultValueDeterminerParameters(parameterData, isOptional, parameterConfig, finalEditor)))
     original.copy(editor = finalEditor, validators = finalValidators, defaultValue = finalDefaultValue)
   }
 
+  private def determineValidatorss(original: Parameter, parameterConfig: ParameterConfig, parameterData: ParameterData, finalEditor: Option[ParameterEditor]) = {
+    // In case if someone forget to specify validators and used only Parameter() or Parameter.optional(), we add missing forgotten validators
+    // It can be tricky, because we have no option to specify that we don't want those "standard" validators
```
Comment (mproch) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala:33]: So, if someone specifies own validators we skip 'standard' ones? Maybe it would be better to add explicitly some flag, like 'skipStandardValidators'? If you need this change now for some reasons - let's add it to changelog/migration guide, as this can be tricky change in behaviour?

**Review by mproch — COMMENTED** [review]
In general I like it, good change. Added some minor remarks

File: utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/ConfluentSchemaRegistryClient.scala (hunk 2 above)
Comment (arkadius) [reply utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/ConfluentSchemaRegistryClient.scala:35]: Because in tests was used cached implementation and each test case was changing behaviour of other test cases

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala (hunk 3 above)
Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala:33]: We currently have such a case: in `GenericTypedJsonSourceFactory` we have parameter `type` with `JsonParameterEditor` which causes adding `JsonValidator` but it is overriden by explicit list of validators. I personally don't like adding those "standard" validators at all and don't want to adding extra fields to `Parameter` for that. WTYT about removing this implication: `JsonParameterEditor` ->  `JsonValidator` and let's developer decide if he/she want to add only editor or both editor and validator?

Comment (mproch) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala:33]: ok, I looked a bit at it, and I'm in favour of not adding these validators at all ;)

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala (hunk 1 above)
Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala]: fixed

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala (hunk 3 above)
Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/parameter/StandardParameterEnrichment.scala]: Changed but for now I only removed `JsonValidator` from those `EditorBasedValidatorExtractor` - must be turned on explicitly because I see a pain for developers to defined both editor and validator for "fixed-values" case. At the end we should rather add something like `ParameterEditor.validators` method and in `Paramter` collect both `validators` and `editor.validators` but don't want to change it in this PR

**Review by mproch — APPROVED** [review]
